ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/moreh_dot_device_operation.cpp

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_dot_device_operation.hpp"
+
+#include "ttnn/tensor/tensor.hpp"
+#include "tt_stl/type_name.hpp"
+#include "ttnn/cpp/ttnn/operations/moreh/moreh_helper_functions.hpp"
+
+namespace ttnn::operations::moreh::moreh_dot_op {
+MorehDotOperation::program_factory_t MorehDotOperation::select_program_factory(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    // For now we litteraly don't care and return a single factory. Whatever
+    return SingleCore{};
+}
+
+void MorehDotOperation::validate_on_program_cache_miss(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+}
+
+void MorehDotOperation::validate_on_program_cache_hit(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+}
+
+void MorehDotOperation::validate(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    const auto& input_tensor_a = tensor_args.input_tensor_a;
+    const auto& input_tensor_b = tensor_args.input_tensor_b;
+
+    TT_ASSERT(tt::operations::primary::is_1d_tensor(input_tensor_a));
```

Comment:
```suggestion
    TT_FATAL(tt::operations::primary::is_1d_tensor(input_tensor_a));
```

I prefer using `TT_FATAL` because `TT_ASSERT` only works in debug mode.
Actually, since validation is not currently executed in Release mode, it doesn't matter at the moment, but we don't know how things might change in the future.